File: Exercises/Exercise1/thread_test.py
from threading import Thread

i = 0

def increase():
	global i

	for j in range(1000000):
		i += 1

def decrease():
	global i

	for j in range(1000000):
		i -= 1

def main():
	thread1 = Thread(target = increase, args = (),)
	thread2 = Thread(target = decrease, args = (),)
	
	thread1.start()
	thread2.start()

	thread1.join()
	thread2.join()

	print i

main()

File: Exercises/Exercise1/thread_test.c
#include <pthread.h>
#include <stdio.h>


void * increase(void);
void * decrease(void);

int i = 0;

int main(void)
{

	pthread_t thread_1;
	pthread_t thread_2;

	pthread_create(&thread_1, NULL, increase, NULL);
	pthread_create(&thread_2, NULL, decrease, NULL);

	pthread_join(thread_1, NULL);
	pthread_join(thread_2, NULL);

	printf("%d\n", i);

	return 0;
}

void * increase(void)
{

	for (int j = 0; j < 1000000; ++j)
	{
		i++;
	}
}

void * decrease(void)
{

	for (int j = 0; j < 1000000; ++j)
	{
		i--;
	}
}
